salt/modules/win_file.py

```diff
@@ -1035,6 +1035,58 @@ def set_mode(path, mode):
     return get_mode(path)
 
 
+def remove(path, force=False):
+    '''
+    Remove the named file or directory
+
+    :param str path: The path to the file or directory to remove.
+
+    :param bool force: Remove even if marked Read-Only
+
+    :return: True if successful, False if unsuccessful
+    :rtype: bool
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' file.remove C:\\Temp
+    '''
+    # This must be a recursive function in windows to properly deal with
+    # Symlinks. The shutil.rmtree function will remove the contents of
+    # the Symlink source in windows.
+    path = os.path.expanduser(path)
+
+    if not os.path.isabs(path):
+        raise SaltInvocationError('File path must be absolute.')
+
+    # Remove ReadOnly Attribute
+    if force:
+        win32api.SetFileAttributes(path, win32con.FILE_ATTRIBUTE_NORMAL)
```

Comment:
https://docs.python.org/2/tutorial/errors.html#defining-clean-up-actions

One option would be to use a `finally` clause to perform any cleanup logic, including resetting perms if necessary. You could also certainly attempt cleanup in the main exception handler. So long as we're not just opening up perms and then walking away in case of an error. :]